packages/databricks-vscode/src/bundle/models/BundleVariableModel.ts

```diff
@@ -0,0 +1,236 @@
+                await this.stateCache.refresh();
+                await this.bundleValidateModel.refresh();
+            }),
+            this.overrideFileWatcher.onDidDelete(async () => {
+                await this.stateCache.refresh();
+                await this.bundleValidateModel.refresh();
+            })
+        );
+    }
+
+    get bundleVariableFilePath() {
+        const target = this.configModel.target;
+        if (target === undefined) {
+            return undefined;
+        }
+
+        return Uri.joinPath(
+            this.workspaceRoot,
+            ".databricks",
+            "bundle",
+            target,
+            "vscode.bundlevars"
+        ).with({scheme: "file"});
+    }
+
+    private async getVariableOverrides(): Promise<Record<string, string>> {
+        if (this.bundleVariableFilePath === undefined) {
+            return {};
+        }
+
+        try {
+            const lines = (
+                await readFile(this.bundleVariableFilePath.fsPath, "utf-8")
+            ).split(os.EOL);
+
+            return Object.fromEntries(
+                lines.map((line) => [
+                    line.split("=")[0],
+                    line.split("=").slice(1).join("="),
+                ])
+            );
+        } catch (e: any) {
+            if (e.code !== "ENOENT") {
+                throw e;
+            }
+            NamedLogger.getOrCreate(Loggers.Extension).debug(
+                "No bundle variable overrides found."
+            );
+            return {};
+        }
+    }
+
+    protected async readState(): Promise<BundleVariableModelState> {
+        const preValidateConfig =
+            await this.configModel.get("preValidateConfig");
+
+        // If no global variables are defined, return an empty map
+        if (preValidateConfig === undefined) {
+            return {};
+        }
+
+        const globalVariableDefinitions =
+            preValidateConfig.preValidateBundleSchema?.variables ?? {};
+        const inTargetVariables = preValidateConfig.variables ?? {};
+        const overrides = await this.getVariableOverrides();
+        const inTargetPostValidateVariables =
+            (await this.configModel.get("validateConfig"))?.variables ?? {};
+
+        const variables: Record<string, BundleVariable> = {};
+
+        for (const [key, definition] of Object.entries(
+            globalVariableDefinitions
+        )) {
+            const inTargetVariable = inTargetVariables[key];
+
+            // We check heuristically if the vairable is required, because bundle validate does not work if
+            // a required variable is not set.
+            const isRequired =
+                (inTargetVariable ??
+                    definition.lookup ??
+                    definition.default) === undefined;
+
+            variables[key] = {
+                ...definition,
+                lookup:
+                    typeof inTargetVariable === "object"
+                        ? inTargetVariable
```

Comment:
Should it be `inTargetVariable.lookup`?

Reply:
No, because when defining new lookups inside a target, users don't have to specify the lookup field.